### DISCUSSION TITLE
Tasmota Station Mode and Connectivity.

### DISCUSSION THREAD

**Discussion by smhmaqsood** [arendst/Tasmota#11516] [Q&A] [answered]
Hi,

I'm facing a problem regarding connectivity and access point mode. I was trying to keep the SoftAP on till it is connected to network. I've gone through a lot of documentation but could not figure out. The main problem is if some had a typo while entering AP SSID and Password then he/she will have to go through device recovery. and I want to keep the softAP on till the device is connected to the network.

**Comment by barbudor** [comment, marked as answer]
Hi
Unless you specified a different WifiConfig as default '2' (such as 4 or 5), tasmota should automatically revert to AP if it can't connect

**Reply by smhmaqsood** [reply]
Thanks for your guidance.
should i change from WIFI_Retry to WIFi_Restart?
I'm not sure about its configuration and how it will work.

**Reply by barbudor** [reply]
My mistake, the default is WIFI_RETRY, so Tasmota will continuously try to connect to the SSID. If wrong you're stuck.
So you should change that to WIFI_MANAGER (2) so Tasmota will restart in WifiManager mode (AP) if it can't connect.
Actually it will restart to SoftAP for 3 minutes, then try again to connect to configured SSID, if not restart in AP for 3 minutes ...
The drawback is that if your wifi gets down, Tasmota will revert to softAP leaving anyone close to connect to your device.

Now the best solution would be to build your own firmware with SSID and Password already defined so that when flashing, you don't have to enter that again. Even in case of recovery it will go to your SSID immediately

**Reply by ascillato** [reply]
With the default settings of Tasmota precompiled binaries:

* if your device has a button configured, push it 6 times to make Tasmota shows the AP to reconfigure.
* if your device don´t have a button, take out the power of the device and power on again 10 times. this will revert all settings to default making Tasmota to show the AP again.
